modules/common/src/main/scala/surge/kafka/KafkaPartitionShardRouterActor.scala

```diff
@@ -161,43 +158,36 @@ class KafkaPartitionShardRouterActor(
 
   // In standby mode, just follow updates to partition assignments and let Kafka streams index the aggregate state
   private def standbyMode(state: ActorState): Receive = {
+    case msg if extractEntityId.isDefinedAt(msg) =>
+      becomeActiveAndDeliverMessage(state, msg)
     case msg: PartitionAssignments     => handle(state, msg)
     case GetPartitionRegionAssignments => sender() ! state.partitionRegions
     case GetHealth =>
       sender() ! HealthCheck(name = "shard-router-actor", id = s"router-actor-$hashCode", status = HealthCheckStatus.UP)
-    case msg if extractEntityId.isDefinedAt(msg) => becomeActiveAndDeliverMessage(state, msg)
   }
 
   private def initialized(state: ActorState): Receive = healthCheckReceiver(state).orElse {
-    case msg: PartitionAssignments               => handle(state, msg)
-    case msg: Terminated                         => handle(state, msg)
-    case GetPartitionRegionAssignments           => sender() ! state.partitionRegions
-    case msg if extractEntityId.isDefinedAt(msg) => deliverMessage(state, msg)
+    case msg: PartitionAssignments     => handle(state, msg)
+    case msg: Terminated               => handle(state, msg)
+    case GetPartitionRegionAssignments => sender() ! state.partitionRegions
+    case msg if extractEntityId.isDefinedAt(msg) =>
+      val entityId = extractEntityId(msg)
+      activeSpan.log("extractEntityId", Map("entityId" -> entityId))
+      deliverMessage(state, entityId, msg)
   }
 
   private def healthCheckReceiver(state: ActorState): Receive = { case GetHealth =>
     getHealthCheck(state).pipeTo(sender())
   }
 
-  private def deliverMessage(state: ActorState, msg: Any): Unit = {
-    Try(extractEntityId(msg)).toOption match {
-      case None =>
-        log.warn("Unsure of how to route message with class [{}], dropping it.", msg.getClass.getName)
-        context.system.deadLetters ! msg
-      case Some(id) =>
-        deliverMessage(state, id, msg)
-    }
-  }
-
   private def deliverMessage(state: ActorState, aggregateId: String, msg: Any): Unit = {
     partitionRegionFor(state, aggregateId) match {
       case Some(responsiblePartitionRegion) =>
-        log.trace(
-          s"RouterActor forwarding command envelope for aggregate $aggregateId to region ${responsiblePartitionRegion.regionManager.pathString}. Msg $msg")
-
-        responsiblePartitionRegion.regionManager.forward(msg)
+        log.trace(s"Forwarding command envelope for aggregate $aggregateId to region ${responsiblePartitionRegion.regionManager.pathString}.")
+        val tracedMsg = TracedMessage(msg, parentSpan = activeSpan)
+        responsiblePartitionRegion.regionManager.forward(tracedMsg)
       case None =>
-        log.error(s"RouterActor could not find a responsible partition region for $aggregateId.")
+        log.error(s"Could not find a responsible partition region for $aggregateId.")
```

Comment:
the log message doesn't need the name of the actor because the name of the actor (or at least the class) should already be shown by logback ...